lowtime/lowtime_rust/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+use pyo3::prelude::*;
+
+use ordered_float::OrderedFloat;
+use pathfinding::directed::edmonds_karp::{
+    SparseCapacity,
+    Edge,
+    edmonds_karp
+};
+
+// for profiling
+use std::time::{
+    Instant,
+    Duration,
+};
+use log::info;
+
+
+#[pyclass]
+struct PhillipsDessouky {
+    #[pyo3(get)]
+    node_ids: Vec<u32>,
+    source_node_id: u32,
+    sink_node_id: u32,
+    edges_raw: Vec<((u32, u32), f64)>,
+}
+
+#[pymethods]
+impl PhillipsDessouky {
+    #[new]
+    fn new(
+        node_ids: Vec<u32>,
+        source_node_id: u32,
+        sink_node_id: u32,
+        edges_raw: Vec<((u32, u32), f64)>,
+    ) -> PyResult<Self> {
+        // Note: Intentionally nothing done here to profile data transfer latency
+
+        // ohjun: do we actually need to return this? This is unncessary data transfer from Rust->Python
+        // Instead, why not have a type that stores a pointer to a PhillipsDessouky, and we can just
+        // send the outer type? This way, Python can use the object to call max_flow and things, but
+        // we can avoid returning all this data here?
+        Ok(PhillipsDessouky {
+            node_ids: node_ids,
+            source_node_id: source_node_id,
+            sink_node_id: sink_node_id,
+            edges_raw: edges_raw,
+        })
+    }
+
+    fn max_flow(&self) -> PyResult<Vec<((u32, u32), f64)>> {
+        let mut profiling_start;
+        let mut profiling_end;
+        // let SCALING_FACTOR: f64 = 1000000000.0;
+
+        profiling_start = Instant::now();
```

Comment:
You can just write `let profiling_start = Instant::now();` everywhere. A new `let` declaration will **shadow** the old variable of the same name. See [this example](https://doc.rust-lang.org/rust-by-example/variable_bindings/scope.html).